apps/web/src/hooks/loadables/useLoadBalances.ts

```diff
@@ -24,46 +24,56 @@ export const useTokenListSetting = (): boolean | undefined => {
   return isTrustedTokenList
 }
 
-export const useLoadBalances = (): AsyncResult<SafeBalanceResponse> => {
-  const [pollCount, resetPolling] = useIntervalCounter(POLLING_INTERVAL)
-  const currency = useAppSelector(selectCurrency)
-  const isTrustedTokenList = useTokenListSetting()
-  const { safe, safeAddress } = useSafeInfo()
+const useCounterfactualBalances = (safe: ExtendedSafeInfo) => {
   const web3 = useWeb3()
   const chain = useCurrentChain()
-  const chainId = safe.chainId
+  const safeAddress = safe.address.value
+  const isCounterfactual = !safe.deployed
+
+  const [data] = useAsync<SafeBalanceResponse | undefined>(() => {
+    if (!chain || !web3 || !isCounterfactual) return
+    return getCounterfactualBalance(safeAddress, web3, chain)
+  }, [chain, safeAddress, web3, isCounterfactual])
 
-  // Re-fetch assets when the entire SafeInfo updates
-  const [data, error, loading] = useAsync<SafeBalanceResponse | undefined>(
-    () => {
-      if (!chainId || !safeAddress || isTrustedTokenList === undefined) return
+  return data as unknown as Balances
+}
 
-      if (!safe.deployed) {
-        return getCounterfactualBalance(safeAddress, web3, chain)
-      }
+export const useRtkBalances = () => {
+  const currency = useAppSelector(selectCurrency)
+  const isTrustedTokenList = useTokenListSetting()
+  const { safe, safeAddress, safeLoaded } = useSafeInfo()
+  const isReady = safeLoaded && isTrustedTokenList !== undefined
 
-      return getBalances(chainId, safeAddress, currency, {
-        trusted: isTrustedTokenList,
-      })
+  const { data, isLoading, error } = useBalancesGetBalancesV1Query(
+    {
+      chainId: safe.chainId,
+      safeAddress,
+      fiatCode: currency,
+      trusted: isTrustedTokenList,
+    },
+    {
+      skip: !isReady,
+      pollingInterval: POLLING_INTERVAL,
     },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [safeAddress, chainId, currency, isTrustedTokenList, pollCount, safe.deployed, web3, chain],
-    false, // don't clear data between polls
   )
 
-  // Reset the counter when safe address/chainId changes
-  useEffect(() => {
-    resetPolling()
-  }, [resetPolling, safeAddress, chainId])
+  // Counterfactual balances
+  const cfData = useCounterfactualBalances(safe)
+
+  return useMemo(
+    () => ({
+      balances: data || cfData || { items: [], fiatTotal: '' },
```

Comment:
nit: I would extract the object fallback into a `defaultBalances` or `emptyBalances` variable